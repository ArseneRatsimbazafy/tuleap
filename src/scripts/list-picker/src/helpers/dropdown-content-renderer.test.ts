/**
 * Copyright (c) Enalean, 2020 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import {
    appendGroupedOptionsToSourceSelectBox,
    appendSimpleOptionsToSourceSelectBox,
} from "../test-helpers/select-box-options-generator";
import { renderListPickerDropdownContent } from "./dropdown-content-renderer";

describe("dropdown-content-renderer", () => {
    let select: HTMLSelectElement, dropdown: Element;

    beforeEach(() => {
        select = document.createElement("select");
        dropdown = document.createElement("span");
        const list = document.createElement("ul");
        list.classList.add("list-picker-dropdown-values-list");
        dropdown.appendChild(list);
    });

    it("renders grouped list items", () => {
        appendGroupedOptionsToSourceSelectBox(select);
        renderListPickerDropdownContent(select, dropdown);

        expect(dropdown.innerHTML).toMatchSnapshot();
    });

    it("renders simple list items", () => {
        appendSimpleOptionsToSourceSelectBox(select);
        renderListPickerDropdownContent(select, dropdown);

        expect(dropdown.innerHTML).toMatchSnapshot();
    });

    it("when the source option is disabled, then the list item should be disabled", () => {
        const disabled_option = document.createElement("option");
        disabled_option.setAttribute("disabled", "disabled");
        disabled_option.setAttribute("value", "You can't select me");

        select.appendChild(disabled_option);

        renderListPickerDropdownContent(select, dropdown);

        const disabled_list_item = dropdown.querySelector(
            ".list-picker-dropdown-option-value-disabled"
        );

        expect(disabled_list_item).not.toBeNull();
    });
});
