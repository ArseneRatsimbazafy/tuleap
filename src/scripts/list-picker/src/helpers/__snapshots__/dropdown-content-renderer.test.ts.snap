// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown-content-renderer renders grouped list items 1`] = `
<ul class="list-picker-dropdown-values-list">
  <li><strong class="list-picker-group-label"></strong>
    <ul role="group" aria-label="Group 1" class="list-picker-dropdown-option-group">
      <li id="item-0" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 0</li>
      <li id="item-1" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 1</li>
      <li id="item-2" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 2</li>
    </ul>
  </li>
  <li><strong class="list-picker-group-label"></strong>
    <ul role="group" aria-label="Group 2" class="list-picker-dropdown-option-group">
      <li id="item-3" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 3</li>
      <li id="item-4" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 4</li>
      <li id="item-5" role="option" aria-selected="false" class="list-picker-dropdown-option-value-disabled">Value 5</li>
    </ul>
  </li>
</ul>
`;

exports[`dropdown-content-renderer renders simple list items 1`] = `
<ul class="list-picker-dropdown-values-list">
  <li id="item-0" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 0</li>
  <li id="item-1" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 1</li>
  <li id="item-2" role="option" aria-selected="false" class="list-picker-dropdown-option-value">Value 2</li>
</ul>
`;
